torch/jit/_recursive.py

```diff
@@ -229,6 +229,13 @@ def infer_type(name, item):
 
         # If we got here, this is a regular "data" attribute, Add it to the concrete type
         attr_type = infer_type(name, value)
+        if attr_type is None and hasattr(value, '__to_ivalue__'):
+            # Manual override of what the JIT conversion should do.
+            #
+            # Kinda a hack: convert the attribute just to get the type. It would
+            # be better if we didnt have to do this, but oh well
+            converted_value = value.__to_ivalue__()
```

Comment:
potentially: can we require that `__to_ivalue__` be annotated with a return type, then query that? Otherwise we will pay a copy cost an additional time.